crates/ruff_python_formatter/src/comments/placement.rs

```diff
@@ -1,13 +1,463 @@
+fn handle_in_between_excepthandlers_or_except_handler_and_else_or_finally_comment<'a>(
+    comment: DecoratedComment<'a>,
+    locator: &Locator,
 ) -> CommentPlacement<'a> {
+    if comment.text_position().is_end_of_line() || comment.following_node().is_none() {
+        return CommentPlacement::Default(comment);
+    }
+
+    if let Some(AnyNodeRef::ExcepthandlerExceptHandler(except_handler)) = comment.preceding_node() {
+        // it now depends on the indentation level of the comment if it is a leading comment for e.g.
+        // the following `elif` or indeed a trailing comment of the previous body's last statement.
+        let comment_indentation =
+            whitespace::indentation_at_offset(locator, comment.slice().range().start())
+                .map(str::len)
+                .unwrap_or_default();
+
+        if let Some(except_indentation) =
+            whitespace::indentation(locator, except_handler).map(str::len)
+        {
+            return if comment_indentation <= except_indentation {
+                // It has equal, or less indent than the `except` handler. It must be a comment
+                // of the following `finally` or `else` block
+                //
+                // ```python
+                // try:
+                //     pass
+                // except Exception:
+                //     print("noop")
+                // # leading
+                // finally:
+                //     pass
+                // ```
+                // Attach it to the `try` statement.
+                CommentPlacement::dangling(comment.enclosing_node(), comment)
+            } else {
+                // Delegate to `handle_trailing_body_comment`
+                CommentPlacement::Default(comment)
+            };
+        }
+    }
+
+    CommentPlacement::Default(comment)
+}
+
+/// Handles comments between the last statement and the first statement of two bodies.
+///
+/// ```python
+/// if x == y:
+///     pass
+///     # This should be a trailing comment of `pass` and not a leading comment of the `print`
+///     # in the `else` branch
+/// else:
+///     print("I have no comments")
+/// ```
+fn handle_in_between_bodies_comment<'a>(
+    comment: DecoratedComment<'a>,
+    locator: &Locator,
+) -> CommentPlacement<'a> {
+    use ruff_python_ast::prelude::*;
+
+    // The rule only applies to own line comments. The default logic associates end of line comments
+    // correctly.
+    if comment.text_position().is_end_of_line() {
+        return CommentPlacement::Default(comment);
+    }
+
+    // The comment must be between two statements...
+    if let (Some(preceding), Some(following)) = (comment.preceding_node(), comment.following_node())
+    {
+        // ...and the following statement must be the first statement in an alternate body of the parent...
+        let is_following_first_in_body = match comment.enclosing_node() {
```

Comment:
this name had me confused for a while because i read it as "is this comment following the first [statement] in the [next] body"

Reply:
I made the name (very) explicit. I hope that helps `is_following_the_first_statement_in_the_parents_alternate_body`